libvast/src/type.cpp

```diff
@@ -686,20 +700,49 @@ detail::generator<std::string_view> type::names() const& noexcept {
   while (true) {
     switch (root->type_type()) {
       case fbs::type::Type::NONE:
+        co_yield "none";
+        co_return;
       case fbs::type::Type::bool_type:
+        co_yield "bool";
+        co_return;
       case fbs::type::Type::integer_type:
+        co_yield "int";
+        co_return;
       case fbs::type::Type::count_type:
+        co_yield "count";
+        co_return;
       case fbs::type::Type::real_type:
+        co_yield "real";
+        co_return;
       case fbs::type::Type::duration_type:
+        co_yield "duration";
+        co_return;
       case fbs::type::Type::time_type:
+        co_yield "time";
+        co_return;
       case fbs::type::Type::string_type:
+        co_yield "string";
+        co_return;
       case fbs::type::Type::pattern_type:
+        co_yield "pattern";
+        co_return;
       case fbs::type::Type::address_type:
+        co_yield "addr";
+        co_return;
       case fbs::type::Type::subnet_type:
+        co_yield "subnet";
+        co_return;
       case fbs::type::Type::enumeration_type:
+        co_yield "enum";
+        co_return;
       case fbs::type::Type::list_type:
+        co_yield "list";
+        co_return;
       case fbs::type::Type::map_type:
+        co_yield "map";
+        co_return;
       case fbs::type::Type::record_type:
+        co_yield "record";
```

Comment:
The giant `switch` feels like it could be factored into a helper function that can be both called from `name()` and `names()`.

Reply:
Actually this was not possible. I reduced it somewhat by naming the constants as `T::kind` for all `concrete_type T`.